subsys/net/ip/net_if.c

```diff
@@ -461,17 +461,6 @@ static void dad_timeout(struct k_work *work)
 		 * in this case as the address is our own one.
 		 */
 		net_ipv6_nbr_rm(iface, &ifaddr->address.in6_addr);
-
-		/* We join the allnodes group from here so that we have
-		 * a link local address defined. If done from ifup(),
-		 * we would only get unspecified address as a source
-		 * address. The allnodes multicast group is only joined
-		 * once in this case as the net_ipv6_mcast_join() checks
-		 * if we have already joined.
-		 */
-		join_mcast_allnodes(iface);
-
-		join_mcast_solicit_node(iface, &ifaddr->address.in6_addr);
```

Comment:
Sounds fine to me, but maybe you'll need to address the case when DAD is a failure: what to do with the mcast then? (or is it relevant actually).
Before, because it was joining mcast only on success, failure case did not have to handle any of it, but now it's different.

Reply:
That's addressed in the commit message:

> This is achieved by joining
from net_if_ipv6_addr_add() call. Note that we already leave
solicited-node group from net_if_ipv6_addr_rm(). In particular, we
leave it if DAD fails (as that function is called in this case).

Reply:
indeed line 745 yes